plugin/trino-datasketches/src/main/java/io/trino/plugin/datasketches/state/SketchStateFactory.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.datasketches.state;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.trino.array.ObjectBigArray;
+import io.trino.spi.function.AccumulatorStateFactory;
+import io.trino.spi.function.GroupedAccumulatorState;
+import org.apache.datasketches.memory.Memory;
+import org.apache.datasketches.theta.SetOperation;
+import org.apache.datasketches.theta.Sketch;
+import org.apache.datasketches.theta.Union;
+import org.openjdk.jol.info.ClassLayout;
+
+public class SketchStateFactory
+        implements AccumulatorStateFactory<SketchState>
+{
+    @Override
+    public SketchState createSingleState()
+    {
+        return new SingleSketchState();
+    }
+
+    @Override
+    public Class<? extends SketchState> getSingleStateClass()
+    {
+        return SingleSketchState.class;
+    }
+
+    @Override
+    public SketchState createGroupedState()
+    {
+        return new GroupedSketchState();
+    }
+
+    @Override
+    public Class<? extends SketchState> getGroupedStateClass()
+    {
+        return GroupedSketchState.class;
+    }
+
+    public static class GroupedSketchState
```

Comment:
Since they are more details to capture, can we move it to a new file ? WDYT ?

Reply:
I think keeping the structure consistent with other state factories would make more sense. Ex. HyperLogLog, LongApproximateMostFrequent, NumericHistogram etc.